crates/bevy_ecs/src/query/filter.rs

```diff
@@ -319,14 +337,25 @@ unsafe impl<T> ReadOnlyFetch for WithoutFetch<T> {}
 /// }
 /// # print_cool_entity_system.system();
 /// ```
+#[derive(Clone, Copy)]
 pub struct Or<T>(pub T);
 
 /// The [`Fetch`] of [`Or`].
+#[derive(Copy)]
 pub struct OrFetch<T: FilterFetch> {
     fetch: T,
     matches: bool,
 }
 
+impl<T: FilterFetch + Clone> Clone for OrFetch<T> {
+    fn clone(&self) -> Self {
+        Self {
+            fetch: self.fetch.clone(),
+            matches: self.matches,
+        }
+    }
+}
```

Comment:
Is this manual impl necessary here?
`T` still get the `Clone` bound and `matches`/`bool` is copy.

I think a `Clone` derive is better here.